### PR TITLE
Cr fix

### PR DESCRIPTION
**Pull request by sushi31415926**

### CODE REVIEW

File: object_tracker/README.md
Comment (asitako1mitt) [inline object_tracker/README.md:19]: Maybe more readable as 'Run Tests + Coverage'

File: object_tracker/README.md
Comment (asitako1mitt) [inline object_tracker/README.md:26]: Maybe more readable as 'Patch PreciseRoIPooling Faster Loading'

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py:7]: Add a 'TODO' note for us to fix Hydra, so we'll configure parameters better. 
In addition, for future work - try to figure out a better way to work with multiple parameterizations (i.e., PyTracking dimp18.py file with all of its params, and the parameters used by the OF and ORB.

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py:8]: Not critical at all... but maybe it will be nice to improve the documentation a bit:
1. missing (s) - sometimes an 's' is missing (set->sets, calculate -> calculates, ...). 
2. A bit more explanation (roi -> initial bounding box, ...). 
As is requires zero-very low efforts, it might be better before merging.

File: object_tracker/README.md
Comment (asitako1mitt) [inline object_tracker/README.md:26]: Maybe more readable as 'Patch PreciseRoIPooling Faster Loading'

File: object_tracker/README.md
Comment (asitako1mitt) [inline object_tracker/README.md]: Maybe more readable as 'Run Tests + Coverage'

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py]: I would change roi to be a namedtuple. It will make things more readable, especially in '_calc_new_roi'.
Suggested names (your call, but things I find reasonable): top_left_x, top_left_y, width, height. 
As there is no requirement for it to be a list (and if so, can be easily converted using the 'list' method), it is a change worth making.

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py:40]: I think it might be good to add to the log a successful initiation and of what BBOX.

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py:53]: I think it is a good thing to log/record a situation where no features are found. 
This is highly unlikely, and can aid a lot in debugging.

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py]: You don't use this variable. I think it might be used for debugging purposes, so please increase it when you finish processing a frame so we will be able to tell what happened when we debug using the logs.

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py:9]: I think its best to get a logger when a matcher is initiated, and log using it.

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py:97]: I would add the frame number for now (at least, maybe even more details in the future) to the log message of the failure. It can help us for debugging when needed.

**Review by asitako1mitt — COMMENTED** [review]
**Partial Review**

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py:96]: For readability purposes, extract p0[st > 0] and p1[st > 0] to variables.

File: object_tracker/match.py
Comment (asitako1mitt) [inline object_tracker/match.py]: Naming can be changed to top_left, top_right,...

File: object_tracker/README.md
Comment (sushi31415926) [reply object_tracker/README.md]: done

File: object_tracker/match.py
Comment (sushi31415926) [reply object_tracker/match.py:53]: done

File: object_tracker/match.py
Comment (sushi31415926) [reply object_tracker/match.py:97]: i deleted it

File: object_tracker/match.py
Comment (asitako1mitt) [reply object_tracker/match.py]: Although it works even if the roi is invalid, it might be smarter to make sure the roi is bounded by the image shape.

File: object_tracker/match.py
Comment (sushi31415926) [reply object_tracker/match.py:9]: i deleted it

File: object_tracker/match.py
Comment (sushi31415926) [reply object_tracker/match.py:40]: i add a log with the method init finish

File: object_tracker/object_tracker.py
Comment (asitako1mitt) [inline object_tracker/object_tracker.py:41]: I would make it more robust to a case where self._tracker_run_iter is set to 0 (and 1). Currently it will crash.

File: object_tracker/match.py
Comment (asitako1mitt) [reply object_tracker/match.py:97]: Alternatively, create a debug mode set to False. but when set to true, saves more info about the failures (e.g., saves the failure frames to a directory, stating the type of failure in the image name or in the debugger so we will be able to match a failure to a frame using any identifier you want like time/frame number/...)

File: object_tracker/object_tracker.py
Comment (asitako1mitt) [inline object_tracker/object_tracker.py:56]: Maybe add the bounding box shape/coordinates as well.

File: tests/test.py
Comment (asitako1mitt) [inline tests/test.py:19]: I suggest dividing it into several tests:
1. Simulate with a regular video.
2. Simulate failed OF (using ORB instead). 
3. Simulate with video with randomly black frames. 
4. Simulate with video with randomly random frames instead of images (or very noise frames once in a while).

**Review by asitako1mitt — CHANGES_REQUESTED** [review]
Finished review.

File: object_tracker/match.py
Comment (sushi31415926) [reply object_tracker/match.py]: done

File: tests/test.py
Comment (sushi31415926) [reply tests/test.py:19]: done, except black frames which i not think its it neccessay

File: object_tracker/match.py
Comment (sushi31415926) [reply object_tracker/match.py:96]: i not think its necessary it is just indexing

File: object_tracker/match.py
Comment (sushi31415926) [reply object_tracker/match.py:8]: done

File: object_tracker/README.md
Comment (sushi31415926) [reply object_tracker/README.md:26]: done

File: object_tracker/object_tracker.py
Comment (sushi31415926) [reply object_tracker/object_tracker.py:41]: done

File: object_tracker/object_tracker.py
Comment (sushi31415926) [reply object_tracker/object_tracker.py:56]: done

**Pull request merged**
